### PR TITLE
Move the stack to the evaluator. (no-op PR for 70598)

### PR DESCRIPTION
**Pull request by vakaras**
The changes to Miri to make it to compile with Rustc PR https://github.com/rust-lang/rust/pull/70598.

### CODE REVIEW

File: src/machine.rs [hunk 1]
```diff
@@ -241,13 +241,16 @@ pub struct Evaluator<'tcx> {
     /// The "time anchor" for this machine's monotone clock (for `Instant` simulation).
     pub(crate) time_anchor: Instant,
 
+    /// The virtual call stack.
+    pub(crate) stack: Vec<Frame<'mir, 'tcx, Tag, FrameData<'tcx>>>,
```
Comment (RalfJung) [inline src/machine.rs:245]: I am not sure what is "virtual" about this -- it's certainly not more virtual than anything else Miri does. ;)

File: src/machine.rs [hunk 2]
```diff
@@ -271,12 +274,13 @@ impl<'tcx> Evaluator<'tcx> {
             panic_payload: None,
             time_anchor: Instant::now(),
             layouts,
+            stack: Default::default(),
```
Comment (RalfJung) [inline src/machine.rs:277]: I'd prefer `Vec::default()` or `Vec::new()` as they give an indication about the type that is being instantiated.

File: src/machine.rs [hunk 3]
```diff
@@ -314,6 +318,20 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
...
+    ) -> &'a [Frame<'mir, 'tcx, Self::PointerTag, Self::FrameExtra>] {
+        &ecx.machine.stack
+    }
+
+    #[inline(always)]
+    fn stack_mut<'a>(
+        ecx: &'a mut InterpCx<'mir, 'tcx, Self>,
+    ) -> &'a mut Vec<Frame<'mir, 'tcx, Self::PointerTag, Self::FrameExtra>> {
+        &mut ecx.machine.stack
+    }
```
Comment (RalfJung) [inline src/machine.rs:333]: Please move these to the same position in the instance as they are in the trait definition.

Comment (vakaras) [reply src/machine.rs]: Done.

**Comment by RalfJung** [comment]
The rustc PR landed! I updated the rustc-version file here.
@bors r+

**Pull request merged**